tools/azsdk-cli/Azure.Sdk.Tools.Cli/Helpers/CodeownersHelper.cs

```diff
@@ -0,0 +1,587 @@
+                {
+                    lines.Insert(index + 1, "");
+                }
+            }
+            else
+            {
+                // If adding at the end, ensure proper spacing
+                if (lines.Count > 0 && !string.IsNullOrWhiteSpace(lines[lines.Count - 1]))
+                {
+                    lines.Add("");
+                }
+                lines.Add(formattedEntry);
+            }
+
+            return string.Join('\n', lines);
+        }
+
+        public string FormatCodeownersEntry(CodeownersEntry codeownersEntry)
+        {
+            var lines = new List<string>();
+
+            bool addSeperationLine = false;
+            bool addEndLine = false;
+
+            string path = codeownersEntry.PathExpression ?? string.Empty;
+            List<string> serviceLabels = codeownersEntry.ServiceLabels ?? new List<string>();
+            List<string> prLabels = codeownersEntry.PRLabels ?? new List<string>();
+            List<string> serviceOwners = codeownersEntry.ServiceOwners ?? new List<string>();
+            List<string> sourceOwners = codeownersEntry.SourceOwners ?? new List<string>();
+            List<string> azureSDKOwners = codeownersEntry.AzureSdkOwners ?? new List<string>();
+
+            // Add all PRLabels first (each on its own line) - derived from service labels
+            if (prLabels.Any())
+            {
+                // ensure label is prefixed without duplicate %
+                var formattedPRLabels = prLabels
+                    .Where(lbl => !string.IsNullOrWhiteSpace(lbl))
+                    .Select(lbl => lbl.StartsWith("%") ? lbl : $"%{lbl}");
+                lines.Add($"# PRLabel: {string.Join(" ", formattedPRLabels)}");
+                addSeperationLine = true;
+            }
+
+            // Add the path and source owners line
+            if (!string.IsNullOrEmpty(path) && sourceOwners != null && sourceOwners.Count > 0)
+            {
+                addSeperationLine = true;
+                // Normalize and deduplicate source owners while preserving original casing
+                var normalizedSourceOwners = sourceOwners
+                    .Where(o => !string.IsNullOrWhiteSpace(o))
+                    .Select(o => o.Trim())
+                    .Select(o => o.StartsWith("@") ? o.Substring(1) : o)
+                    .Select(o => o.Trim())
+                    .ToList();
+
+                var uniqueSourceOwners = new List<string>();
+                var seenSource = new HashSet<string>(StringComparer.OrdinalIgnoreCase);
+                foreach (var owner in normalizedSourceOwners)
+                {
+                    if (seenSource.Add(owner))
+                    {
+                        uniqueSourceOwners.Add("@" + owner);
+                    }
+                }
+
+                var pathLine = $"{path}".PadRight(67) + $"{string.Join(" ", uniqueSourceOwners)}";
```

Comment:
Why 67?

Reply:
This code should probably live in the codeowners utils project then so the behavior is consistent in all usages.